File: src/commonMain/kotlin/mahjongutils/models/hand/Hand.kt
package mahjongutils.models.hand

import kotlinx.serialization.Serializable
import mahjongutils.models.Furo
import mahjongutils.models.Tile

/**
 * 手牌
 */
@Serializable
data class Hand<out P : HandPattern>(
    /**
     * 门前的牌
     */
    val tiles: List<Tile>,
    override val furo: List<Furo>,
    /**
     * 手牌形
     */
    val patterns: Collection<P>
) : IHasFuro

File: src/models/hand/Hand.ts
import { type Furo } from '../Furo'
import { type AbstractHandPattern, type HandPattern } from './HandPattern'
import { type Tile } from '../Tile'

export interface Hand<P extends HandPattern = AbstractHandPattern> {
  tiles: Tile[]
  furo: Furo[]
  patterns: P[]
}
